### PR TITLE
Update `util` extern to v10

### PR DESCRIPTION
**Pull request by terurou**
If this PR is merged, `util` extern becomes completely v10.

Changes:

* Update `util.Inspect`
    * https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_util_inspect_custom
* Add `util.TextDecoder`
    * https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_class_util_textdecoder
* Add `util.TextEncoder`
    * https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_class_util_textencoder
* Add `util.Types`
    * https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_util_types
* Fix coding style

### CODE REVIEW

File: src/js/node/util/Types.hx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+	Unlike `Syntax.instanceof` or `Object.prototype.toString.call(value)`, these checks do not
+	inspect properties of the object that are accessible from JavaScript (like their prototype),
+	and usually have the overhead of calling into C++.
+
+	The result generally does not make any guarantees about what kinds of properties or behavior
+	a value exposes in JavaScript. They are primarily useful for addon developers who prefer to do
+	type checking in JavaScript.
+**/
+@:jsRequire("util", "types")
+extern class Types {
```
Comment (Gama11) [inline src/js/node/util/Types.hx]: Shouldn't all the methods here be declared as `static`?

File: src/js/node/util/TextDecoder.hx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	**/
+	var ignoreBOM(default,null):Bool;
+}
+
+typedef TextDecoderOptions = {
+	/**
+		`true` if decoding failures are fatal. This option is only supported when ICU is enabled (see Internationalization).
+		
+		Default: `false`.
+	 */
```
Comment (Gama11) [inline src/js/node/util/TextDecoder.hx]: A few doc comments here end with `*/` instead of `**/`.

File: src/js/node/util/Types.hx (hunk 1 above)
Comment (terurou) [reply src/js/node/util/Types.hx]: Thnaks for your review comment! I fixed it.

File: src/js/node/util/TextDecoder.hx (hunk 2 above)
Comment (terurou) [reply src/js/node/util/TextDecoder.hx]: Thanks, I fixed it.
If you are using the linter, please tell me.

**Comment by Gama11** [comment]
Thanks! Awesome to get so many contributions suddenly, keep it up. ;)

**Pull request merged**